File: sdks/python/cyanprintsdk/api/resolver/req.py
from typing import Any, Dict, List

from cyanprintsdk.api.base_model import CyanBaseModel


class FileOriginReq(CyanBaseModel):
    template: str
    layer: int


class ResolvedFileReq(CyanBaseModel):
    path: str
    content: str
    origin: FileOriginReq


class ResolverReq(CyanBaseModel):
    config: Dict[str, Any]
    files: List[ResolvedFileReq]

File: sdks/node/src/api/resolver/req.ts
interface FileOriginReq {
  template: string;
  layer: number;
}

interface ResolvedFileReq {
  path: string;
  content: string;
  origin: FileOriginReq;
}

interface ResolverReq {
  config: Record<string, unknown>;
  files: ResolvedFileReq[];
}

export type { FileOriginReq, ResolvedFileReq, ResolverReq };
